mysql/infoschema.go

```diff
@@ -304,6 +310,78 @@ func getConstraints(conv *internal.Conv, db *sql.DB, table schemaAndName) ([]str
 	return primaryKeys, m, nil
 }
 
+// getForeignKeys return list all the foreign keys constraints.
+func getForeignKeys(conv *internal.Conv, db *sql.DB, table schemaAndName) (foreignKeys []schema.Fkey, err error) {
+	refTables, err := getRefTables(db, table)
+	if err != nil {
+		return nil, err
+	}
+	for _, refTable := range refTables {
+		fkey, err := getForeignKey(conv, db, table, refTable)
+		if err != nil {
+			return nil, err
+		}
+		foreignKeys = append(foreignKeys, fkey)
+	}
+	return foreignKeys, nil
+}
+
+// getRefTables return the list of referenced tables for
+// the selected database and table.
+func getRefTables(db *sql.DB, table schemaAndName) ([]schemaAndName, error) {
+	q := "select REFERENCED_TABLE_NAME FROM INFORMATION_SCHEMA.REFERENTIAL_CONSTRAINTS WHERE CONSTRAINT_SCHEMA=? and TABLE_NAME=?;"
+	rows, err := db.Query(q, table.schema, table.name)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't get reference tables: %w", err)
+	}
+	defer rows.Close()
+	var refTableName string
+	var refTables []schemaAndName
+	for rows.Next() {
+		rows.Scan(&refTableName)
+		refTables = append(refTables, schemaAndName{schema: table.schema, name: refTableName})
+	}
+	return refTables, nil
+}
+
+// getForeignKey returns the foreign key constraint for
+// a particular referenced table in the selected table
+// and database.
+func getForeignKey(conv *internal.Conv, db *sql.DB, table schemaAndName, refTable schemaAndName) (schema.Fkey, error) {
+	q := `SELECT k.COLUMN_NAME,k.REFERENCED_COLUMN_NAME 
+		FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS AS t 
+		INNER JOIN INFORMATION_SCHEMA.KEY_COLUMN_USAGE AS k 
+			ON t.CONSTRAINT_NAME = k.CONSTRAINT_NAME 
+			AND t.CONSTRAINT_SCHEMA = k.CONSTRAINT_SCHEMA 
+			AND t.TABLE_NAME = k.TABLE_NAME 
+		WHERE k.TABLE_SCHEMA = ? 
+			AND k.TABLE_NAME = ? 
+			AND t.CONSTRAINT_TYPE = "FOREIGN KEY" 
+			AND k.REFERENCED_TABLE_NAME = ? 
+		ORDER BY
+			k.REFERENCED_TABLE_NAME,
+			k.ordinal_position;`
+	rows, err := db.Query(q, table.schema, table.name, refTable.name)
```

Comment:
Clarifying question: so does this mean that the refTable and table have to have the same schema i.e. I can't have foreign key constraints across schemas in mysql.

Reply:
I think we had discussed this long back that in mysql, schema is same as database. We are already selecting a database for migration which translates to all the constraints in same schema only. While it's not the case with postgres ans hence it's handled a bit differently in postgres with addition of `refTable.schema` argument.

Reply:
But it seems it is possible to have foreign key constraints across databases in mysql (https://forums.mysql.com/read.php?22,150829,200251#msg-200251).
So that would translate to refTable having a different schema here?

Aside: I find it a little mind-blowing that mysql supports this ... maybe it is simplest for us to ignore it (not support it)?
In which case, maybe a comment in code?

Reply:
Yes Nevin, you are absolutely correct. That's the reason why I have not added support for foreign key in other schema because at the end it will not have any significance and will point nowhere. Schema in MySQL is same as Database contrary to Postgres.

Sure, I will add a comment and the constraint to force it.